pkg/template/controller/templateinstance_controller.go

```diff
@@ -407,7 +422,7 @@ func (c *TemplateInstanceController) instantiate(templateInstance *templateapi.T
 		return err
 	}
 
-	errs := runtime.DecodeList(template.Objects, legacyscheme.Codecs.UniversalDecoder())
+	errs := runtime.DecodeList(template.Objects, legacyscheme.Codecs.LegacyCodec(schema.GroupVersion{Group: "", Version: "v1"}), unstructured.UnstructuredJSONScheme)
```

Comment:
That looks suspicious.  Seems like everything should be groupified.

Reply:
This processes arbitrary and legacy templates. Not everything will be groupified.

Reply:
@liggitt can fill you in on why we did all this magic.  I think the short version is:

1) ideally we want everything to just be Unstructured
2) but if it's unstructured, then our mapper doesn't know how to send it to the right api when it's a legacy openshift api object (e.g. non-groupified Route objects end up getting sent to the k8s api endpoint)
3) therefore we first decode all legacy openshift objects into their real types, then decode everything else as unstructured

Reply:
> This processes arbitrary and legacy templates. Not everything will be groupified.

So switching process to groupify on the way out would resolve this for all callers?  Seems like that may be easier.

Reply:
Yeah, this was specifically to get typed objects for v1 so existing /oapi differentiators would be effective.